### PR TITLE
Update custom-ai-function.rst

### PR DESCRIPTION
**Pull request by IshSiva**
updated the steps to create a new AI function with EvaDB.

### CODE REVIEW

File: docs/source/reference/ai/custom-ai-function.rst [hunk 1]
```diff
@@ -39,47 +41,102 @@ An abstract method that must be implemented in your child class. The setup funct
   - True: Batching should be enabled
   - False: Batching is disabled.
 
-The custom setup operations for the function can be written inside the function in the child class. If there is no need for any custom logic, then you can just simply write "pass" in the function definition.
-
-Example of a Setup Function
+Any additional arguments needed for creating the function must be passed as arguments to the setup function. *(Please refer to the ChatGPT function example)*
```
Comment (xzdandy) [inline docs/source/reference/ai/custom-ai-function.rst:44]: Can we give a link to the `ChatGPT function example`?

File: docs/source/reference/ai/custom-ai-function.rst [hunk 2]
```diff
@@ -39,47 +41,102 @@ An abstract method that must be implemented in your child class. The setup funct
...
 
-.. code-block:: python
+The custom setup operations for the function can be written inside the function in the child class. If there is no need for any custom logic, then you can just simply write "pass" in the function definition.
 
-  @setup(cacheable=True, function_type="object_detection", batchable=True)
-  def setup(self, threshold=0.85):
-      #custom setup function that is specific for the function
-      self.threshold = threshold 
-      self.model = torch.hub.load("ultralytics/yolov5", "yolov5s", verbose=False)
```
Comment (xzdandy) [inline docs/source/reference/ai/custom-ai-function.rst:52]: I think we should give an example here. What do you think?

File: docs/source/reference/ai/custom-ai-function.rst [hunk 3]
```diff
@@ -39,47 +41,102 @@ An abstract method that must be implemented in your child class. The setup funct
...
    
   Data types of the inputs to the forward function must be specified. If no constraints are given, then no validation is done for the inputs.
 
-- output_signatures: List[IOColumnArgument]
+- output_signatures: List[IOArgument]
 
   Data types of the outputs to the forward function must be specified. If no constraints are given, then no validation is done for the inputs.
 
-A sample forward function is given below
+A list of IO types in EvaDB are found in `IODescriptors <https://github.com/georgia-tech-db/evadb/blob/staging/evadb/functions/decorators/io_descriptors/data_types.py>`_ folder.
```
Comment (xzdandy) [inline docs/source/reference/ai/custom-ai-function.rst:62]: Shall we use the link to the master branch instead?

File: docs/source/reference/ai/custom-ai-function.rst [hunk 4]
```diff
@@ -39,47 +41,102 @@ An abstract method that must be implemented in your child class. The setup funct
...
 
-  @setup(cacheable=True, function_type="object_detection", batchable=True)
-  def setup(self, threshold=0.85):
-      #custom setup function that is specific for the function
-      self.threshold = threshold 
-      self.model = torch.hub.load("ultralytics/yolov5", "yolov5s", verbose=False)
 
-Forward
---------
+**Forward**
```
Comment (xzdandy) [inline docs/source/reference/ai/custom-ai-function.rst:49]: I think we shall also give an example for forward function.

Comment (xzdandy) [reply docs/source/reference/ai/custom-ai-function.rst:49]: I see the example below.

File: docs/source/reference/ai/custom-ai-function.rst [hunk 5]
```diff
@@ -39,47 +41,102 @@ An abstract method that must be implemented in your child class. The setup funct
...
     
-    @forward(
-          input_signatures=[
-              PyTorchTensor(
-                  name="input_col",
-                  is_nullable=False,
-                  type=NdArrayType.FLOAT32,
-                  dimensions=(1, 3, 540, 960),
+          self.threshold = threshold #sets the threshold for the model
+          self.model = YOLO(model) #initializes the model
```
Comment (xzdandy) [inline docs/source/reference/ai/custom-ai-function.rst:120]: Where does the `model` variable come from?

File: docs/source/reference/ai/custom-ai-function.rst [hunk 6]
```diff
@@ -39,47 +41,102 @@ An abstract method that must be implemented in your child class. The setup funct
...
+          self.device = "cpu" #sets the device as CPU
```
Comment (xzdandy) [inline docs/source/reference/ai/custom-ai-function.rst:121]: Is GPU case for interesting for the users?

File: docs/source/reference/ai/custom-ai-function.rst [hunk 7]
```diff
@@ -88,64 +145,139 @@ A sample forward function is given below
...
-                lambda x: list(x), axis=1
-            )
-
-            outcome.append(
-                {"labels": pred_class, "bboxes": pred_boxes, "scores": pred_score}
-            )
-
-        return pd.DataFrame(outcome, columns=["labels", "bboxes", "scores"])
+      def forward(self, text_df):
+        <function definition>
```
Comment (xzdandy) [inline docs/source/reference/ai/custom-ai-function.rst:248]: This is a bit strange.

File: docs/source/reference/ai/custom-ai-function.rst (hunk 1 above)
Comment (xzdandy) [inline docs/source/reference/ai/custom-ai-function.rst:44]: We may also want to emphasize that the additional arguments to the `setup` are passed in when user 
`CREATE FUNCTION`. We can link to the documentation for `CREATE FUNCTION` here also.

Comment (IshSiva) [reply docs/source/reference/ai/custom-ai-function.rst]: done

File: docs/source/reference/ai/custom-ai-function.rst (hunk 6 above)
Comment (IshSiva) [reply docs/source/reference/ai/custom-ai-function.rst:124]: I am not sure about this. Since most people run it on the local they mostly use CPU.

**Review by xzdandy — APPROVED** [review]
Thanks for updating the custom AI function documentation!

**Comment by xzdandy** [comment]
Hi @IshSiva, please check https://evadb.readthedocs.io/en/latest/source/reference/ai/custom-ai-function.html, there are several format and indentation issues. Could you help fix? Thanks!

**Pull request merged**